src/dataflow-types/types.rs

```diff
@@ -481,7 +481,7 @@ pub enum ExternalSourceConnector {
 impl ExternalSourceConnector {
     pub fn metadata_columns(&self) -> Vec<(ColumnType, Option<String>)> {
         match self {
-            Self::Kafka(_) => vec![],
+            Self::Kafka(_) => vec![(ColumnType::new(ScalarType::Int64), Some("mz_offset".into()))],
```

Comment:
What is the purpose of the `mz_offset` column? I see we're exposing it to users, is that necessary?

Reply:
Yep, it's intentional! More context here: https://github.com/MaterializeInc/materialize/issues/1577